src/openms/source/FORMAT/MzQCFile.cpp

```diff
@@ -0,0 +1,244 @@
+    json out;
+    // required: creationDate, version
+    DateTime currentTime = DateTime::now();
+    out["mzQC"]["creationDate"] = currentTime.toString();
+    out["mzQC"]["version"] = "1.0.0";
+
+    // optional: contactName, contactAddress, description
+    if (!contactName.empty()) 
+    {
+      out["mzQC"]["contactName"] = contactName;
+    }
+    if (!contactAddress.empty()) 
+    {
+      out["mzQC"]["contactAddress"] = contactAddress;
+    }
+    if (!description.empty()) 
+    {
+      out["mzQC"]["description"] = description;
+    }
+
+    out["mzQC"]["runQualities"]["metadata"]["label"] = label;
+    out["mzQC"]["runQualities"]["metadata"]["inputFiles"] = {
+        {
+          {"location", File::absolutePath(inputFileName)},
+          {"name", File::basename(inputFileName)},
+          {"fileFormat",
+          {
+            {"accession", "MS:10000584"},
+            {"name", "mzML format"}
+          }
+          }
+        },
+          "fileProperties",
+          {
+            {
+              {"accession", "MS:1000747"},
+              {"name", "completion time"},
+              {"value", String(exp.getDateTime().getDate() + "T" + exp.getDateTime().getTime())}
+            },
+            {
+              {"accession", "MS:1000569"},
+              {"name", "SHA-1"},
+              {"value", String(FileHandler::computeFileHash(inputFileName))}
+            },
+            {
+              {"accession", "MS:1000031"},
+              {"name", "instrument model"},
+              {"value", String(exp.getInstrument().getName())}
+            }
+          }
+    };
+
+    VersionInfo::VersionDetails version = VersionInfo::getVersionStruct();
+    out["mzQC"]["runQualities"]["metadata"]["analysisSoftware"] = { // todo
+        {
+          {"accession", "MS:1009001" }, // create new qc-cv for QCCalculator: MS:1009001 quality control metrics generating software
+          {"name", "QCCalculator"},
+          {"version", String(version.version_major)+"."+String(version.version_minor)+"."+String(version.version_patch)},
+          {"uri", "https://www.openms.de"}
+        }
+    };
+
+    out["mzQC"]["runQualities"]["qualityMetrics"] = {};
+    
+    out["mzQC"]["controlledVocabularies"] = {
+        {
+          {"name", "Proteomics Standards Initiative Quality Control Ontology"},
+          {"uri", "https://github.com/HUPO-PSI/mzQC/blob/master/cv/qc-cv.obo"},
+          {"version", "0.1.2"},
+        },
+        {
+          {"name", "Proteomics Standards Initiative Mass Spectrometry Ontology"},
+          {"uri", "https://github.com/HUPO-PSI/psi-ms-CV/blob/master/psi-ms.obo"},
+          {"version", "4.1.49"}
+        }
+    };
+    ControlledVocabulary cv;
+    cv.loadFromOBO("PSI-MS", File::find("/CV/psi-ms.obo"));
+    cv.loadFromOBO("QC", File::find("/CV/qc-cv.obo"));
+
+    // create qualityMetric in json format
+    auto addValue = [&cv](const String& accession, const String& name, const auto& value) -> json
```

Comment:
I passed accession and name because it is a human readable annotation and if something changes in the OBO it can be changed in one place.  It is also possible to just add the name as a code comment and fetch the name from the CV by accession.
In this tool there will be only terms that are officially supported.